pkg/cleanup/cleanup.go

```diff
@@ -73,6 +112,26 @@ func LimitTags(tags *[]string, keep int) []string {
 	return []string{}
 }
 
+// FilterImageTagsByTime returns the tags which are older than the specified time
+func FilterImageTagsByTime(imageStreamObjectTags *[]imagev1.NamedTagEventList, olderThan time.Time) []string {
+	var imageStreamTags []string
+
+	for _, imageStreamTag := range *imageStreamObjectTags {
+		lastUpdatedDate := imageStreamTag.Items[0].Created.Time
```

Comment:
What is the `Items[0]`? Are there never more items?

Reply:
When an image tag is created an object of type `TagEvent` is generated inside `Items` array. This array of `TagEvents` has relevant information about the history of the tag and can be useful to deduct info such as creation date or last update date. There's always at least one TagEvent (the creation of image tag itself) and whenever there is a modification to the tag a new TagEvent is created.

Reply:
I personally feel safer if we sort it explicitly instead of assuming that the first is always the newest. Have you checked the API documentation? Is there a link to API docs that states that it's guaranteed to be latest or was our test just coincidence? If so please add a comment

Reply:
We do not assume anything, that line of code is part of an algorithm, I only assume that is is present and it should be otherwise the image tag is invalid. As agreed with @srueg the reference date for an image is the last update date. In fact if you take a look at the next 5 lines of code, the software is searching the value via a `for` loop. No need for sorting in this case.

Reply:
Ah, missed that one, yes. All good